subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/UnsupportedTypesCodecs.kt

```diff
@@ -135,3 +141,24 @@ fun BindingsBuilder.unsupportedTypes() {
     // Gradle implementation types
     bind(unsupported<DefaultServiceRegistry>())
 }
+
+
+internal
+object UnsupportedFingerprintBuildServiceProviderCodec : Codec<BuildServiceProvider<*, *>> {
+    override suspend fun WriteContext.encode(value: BuildServiceProvider<*, *>) {
+        logUnsupported("serialize",
+            documentationSection = DocumentationSection.NotYetImplementedBuildServiceInFingerprint,
+            unsupportedThings = {
+                text(" BuildServiceProvider of service ")
+                reference(value.serviceDetails.implementationType)
+                text(" with name ")
+                reference(value.serviceDetails.name)
+                text(" used at configuration time")
```

Comment:
🟡 I think _used at configuration time_ is too generic and it might become a source of confusion but I don't have a better suggestion right now.

Reply:
I don't think it is possible to hit this without going through a ValueSource, but I might be missing something here, so I went with a generic term. "Fingerprinting" would be more precise, but we don't have it as a user-facing concept,AFAIK.